cider/tests/nextgen/jitlib/JITLibTest.cpp

```diff
@@ -143,6 +143,31 @@ TEST_F(JITLibTests, LogicalOpTest) {
       false, true, true, [](JITValue& a, JITValue& b) { return !a; });
 }
 
+TEST_F(JITLibTests, ExternalModuleTest) {
+  LLVMJITModule module("Test Module", true);
+
+  JITFunctionPointer function1 = module.createJITFunction(
+      JITFunctionDescriptor{.function_name = "test_externalModule",
+                            .ret_type = JITFunctionParam{.type = JITTypeTag::INT32},
+                            .params_type = {}});
+  {
+    JITValuePointer x = function1->createVariable("x1", JITTypeTag::INT32);
+    JITValuePointer a = function1->createConstant(JITTypeTag::INT32, 123);
+    JITValuePointer b = function1->createConstant(JITTypeTag::INT32, 876);
+    *x = *function1->emitRuntimeFunctionCall(
+        "external_call_test_sum",
+        JITFunctionEmitDescriptor{.ret_type = JITTypeTag::INT32,
+                                  .params_vector = {a.get(), b.get()}});
+    function1->createReturn(*x);
+  }
+
+  function1->finish();
```

Comment:
Since we only have one module in a query fragment, Module builder will be rarely used, it not necessary to provide a builder for module? And I will add a function builder in another PR.